core/tests/protractor_utils/AdminPage.js

```diff
@@ -292,6 +333,154 @@ var AdminPage = function() {
+          'Text not showing up in status message');
+      }
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Similarity upload failed – download similarities button not clickable.');
+  };
+
+  this.downloadSimilarityFile = async function() {
+    await action.click('Download Similarity File Button',
+      similarityDownloadButton);
+  };
+
+  this.clearSearchIndex = async function() {
+    await action.click('Clear search index button', searchIndexClearButton);
+    await general.acceptAlert();
+    await waitFor.visibilityOf(statusMessage,
+      'Status message not showing up.');
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Index successfully cleared.');
+  };
+
+  this.flushMigrationBotContributions = async function() {
+    await action.click('Migration bot flush contributions button',
+      flushMigrationBotContributionsButton);
+    await general.acceptAlert();
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Migration bot contributions successfully flushed.',
+      'Migration bot cuntributions not flushing.');
+  };
+
+  this.regenerateContributionsForTopic =
+    async function(topicId, isMeantToSucceed) {
+      await waitFor.visibilityOf(regenerateContributionsTopicIdInput,
+        'Regenerate contributions topic ID input not showing up.');
+      await regenerateContributionsTopicIdInput.sendKeys(topicId);
+      await action.click('Regenerate contributions form submit button',
+        regenerateContributionsSubmitButton);
+      await general.acceptAlert();
+      await waitFor.visibilityOf(regenerateContributionsTopicIdInput,
+        'Regenerate contributions topic ID input not showing up.');
+      await regenerateContributionsTopicIdInput.clear();
+      var text;
+      if (!isMeantToSucceed) {
+        text = 'Server error: Entity for class TopicModel with id ' +
+          topicId + ' not found';
+      } else {
+        text = 'No. of opportunities model created: 0';
+      }
+      await waitFor.visibilityOf(regenerationMessage,
+        'Regeneration message not visible');
+      await waitFor.textToBePresentInElement(regenerationMessage, text);
+    };
+
+  this.changeUsername = async function(oldUsername, newUsername) {
+    await waitFor.visibilityOf(oldUsernameInput,
+      'Current username input not visible');
+    await oldUsernameInput.sendKeys(oldUsername);
+    await waitFor.visibilityOf(newUsernameInput,
+      'New username input not visible');
+    await newUsernameInput.sendKeys(newUsername);
+    await waitFor.visibilityOf(usernameChangeSubmitButton,
+      'Change username submit button not visible');
+    await action.click('Username change submit button',
+      usernameChangeSubmitButton);
+    var text = 'Successfully renamed ' + oldUsername + ' to ' + newUsername;
+    await waitFor.textToBePresentInElement(statusMessage, text,
```

Comment:
I see that everywhere we are using the same statusMessage for the assertion, so why not have expectStatusMessageToBe method and call it from the describe/it block with the status message? (This will help us avoid duplicate code as well as better test structure!)

Reply:
Also, should we have a check like:

waitForStatusMeassage to be visible
expect(StatusMessage).toBe(...)